code/modules/atmospherics/machinery/components/trinary_devices/hypertorus.dm

```diff
@@ -0,0 +1,893 @@
+	if(instability * 0.5 < FUSION_INSTABILITY_ENDOTHERMALITY)
+		internal_instability = 1
+	else
+		internal_instability = -1
+
+	/*
+	 *Modifiers
+	 */
+	///Those are the scaled gases that gets consumed and releases energy or help increase that energy
+	var/positive_modifiers = scaled_hydrogen + scaled_tritium + scaled_m_nitrogen * 0.35 + scaled_m_co2 * 0.55 + scaled_m_antinoblium * 10 - scaled_m_hypernoblium * 10
+	///Those are the scaled gases that gets produced and consumes energy or help decrease that energy
+	var/negative_modifiers = scaled_helium + scaled_m_h2o * 0.75 + scaled_m_freon * 1.15 - scaled_m_antinoblium * 10
+	///Between 0.25 and 100, this value is used to modify the behaviour of the internal energy and the core temperature based on the gases present in the mix
+	var/power_modifier = clamp(scaled_tritium * 1.05 + scaled_m_co2 * 0.95 + scaled_m_plasma * 0.05 - scaled_helium * 0.55 - scaled_m_freon * 0.75, 0.25, 100)
+	///Minimum 0.25, this value is used to modify the behaviour of the energy emission based on the gases present in the mix
+	var/heat_modifier = max(scaled_hydrogen * 1.15 + scaled_helium * 1.05 + scaled_m_plasma * 1.25 - scaled_m_nitrogen * 0.75 - scaled_m_freon * 0.95, 0.25)
+	///Between 0.005 and 1000, this value modify the radiation emission of the reaction, higher values increase the emission
+	var/radiation_modifier = clamp(scaled_helium * 0.55 - scaled_m_freon * 1.15 - scaled_m_nitrogen * 0.45 - scaled_m_plasma * 0.95 + scaled_m_bz * 1.9 + scaled_m_proto_nitrate * 0.1 + scaled_m_antinoblium * 10, 0.005, 1000)
+
+	/*
+	 *Main calculations (energy, internal power, core temperature, delta temperature,
+	 *conduction, radiation, efficiency, power output, heat limiter modifier and heat output)
+	 */
+	//Can go either positive or negative depending on the instability and the negative_modifiers
+	//E=mc^2 with some changes for gameplay purposes
+	energy += internal_instability * ((positive_modifiers - negative_modifiers) * LIGHT_SPEED ** 2) * max(internal_fusion.temperature / (max(100 / heat_modifier, 1)), 1)
+	//Power of the gas mixture
+	internal_power = (scaled_hydrogen / max((100 / power_modifier), 1)) * (scaled_tritium / max((100 / power_modifier), 1)) * (PI * (2 * (scaled_hydrogen * CALCULATED_H2RADIUS) * (scaled_tritium * CALCULATED_TRITRADIUS))**2) * energy
+	//Temperature inside the center of the gas mixture
+	core_temperature = internal_power / max((1000 / power_modifier), 1)
+	core_temperature = max(TCMB, core_temperature)
+	//Difference between the gases temperature and the internal temperature of the reaction
+	delta_temperature = archived_heat - core_temperature
+	//Energy from the reaction lost from the molecule colliding between themselves.
+	conduction = - delta_temperature * (magnetic_constrictor / 10)
+	//The remaining wavelength that actually can do damage to mobs.
+	radiation = max(- (PLANCK_LIGHT_CONSTANT / (((0.0005) * 1e-14) / radiation_modifier)) * delta_temperature, 0)
```

Comment:
```suggestion
	radiation = max(-(PLANCK_LIGHT_CONSTANT / 5e-18) * radiation_modifier * delta_temperature, 0)
```
(0.0005 * 1e-14 = 5e18)

My healthspan gets shortened every time I see this. Unless you have a very good reason to do this, please simplify this madness. Moreover, the whole `-(PLANCK_LIGHT_CONSTANT / 5e-18)` will evaluate into `-40` probably on compile time, so if you feel extra lazy, you can do that.